tool/util/lifecycle.dart

```diff
@@ -1,7 +1,33 @@
 part of dsa.links.repository.util;
 
-fail(String msg, {String out}) async {
-  print("ERROR: ${msg}");
+dbg(String msg) {
+  if (isDebugOn) {
+    var prefix = "[DEBUG]";
+    if (currentLinkBuild != null) {
+      prefix += "[${currentLinkDisplayName}]";
+    }
+    var lines = msg.trim().split("\n");
+    for (var line in lines) {
+      line = line.trim();
+      if (line.isEmpty || line == ".") {
+        continue;
+      }
+
+      print("${prefix} ${line}");
+    }
+  }
+}
+
+fail(String msg, {String out, bool firstLineOnlyDebug: false}) async {
```

Comment:
Sometimes there are multiple lines of output when we provide command output in failure messages, but when in debug mode, the output would be duplicated. This option allows specifying that we don't want to duplicate it in debug mode.